rotkehlchen/data/importer.py

```diff
@@ -1014,3 +1015,167 @@ def import_shapeshift_trades_csv(self, filepath: Path) -> Tuple[bool, str]:
+                        action_type = LedgerActionType.INCOME
+                    elif transaction_type == 'out':
+                        if destination_asset == A_BAT:
+                            action_type = LedgerActionType.GIFT
+                        else:
+                            action_type = LedgerActionType.EXPENSE
+                    else:
+                        log.debug(f'Ignoring unmapped uphold internal transaction. {csv_row}')
+                        return
+                    action = LedgerAction(
+                        identifier=0,
+                        timestamp=timestamp,
+                        action_type=action_type,
+                        location=Location.UPHOLD,
+                        amount=destination_amount,
+                        asset=destination_asset,
+                        rate=None,
+                        rate_asset=None,
+                        link="",
+                        notes=notes,
+                    )
+                    self.db_ledger.add_ledger_action(action)
+            # Trades (buy)
+            else:
+                if transaction_type == 'transfer':
+                    trade = Trade(
+                        timestamp=timestamp,
+                        location=Location.UPHOLD,
+                        base_asset=destination_asset,
+                        quote_asset=origin_asset,
+                        trade_type=TradeType.BUY,
+                        amount=destination_amount,
+                        rate=Price(origin_amount / destination_amount),
+                        fee=Fee(fee),
+                        fee_currency=fee_asset,
+                        link='',
+                        notes=notes,
+                    )
+                    self.db.add_trades([trade])
+        # Outgoing
+        if origin == 'uphold' and destination != 'uphold' and transaction_type == 'out':
+            # Withdrawal
+            if destination_asset == origin_asset:
+                asset_movement = AssetMovement(
+                    location=Location.UPHOLD,
+                    category=AssetMovementCategory.WITHDRAWAL,
+                    address=None,
+                    transaction_id=None,
+                    timestamp=timestamp,
+                    asset=origin_asset,
+                    amount=origin_amount,
+                    fee=Fee(fee),
+                    fee_asset=fee_asset,
+                    link='',
+                )
+                self.db.add_asset_movements([asset_movement])
+            # Trades (sell)
+            else:
+                trade = Trade(
+                    timestamp=timestamp,
+                    location=Location.UPHOLD,
+                    base_asset=origin_asset,
+                    quote_asset=destination_asset,
+                    trade_type=TradeType.SELL,
+                    amount=origin_amount,
+                    rate=Price(destination_amount / origin_amount),
```

Comment:
If `origin_amount` is ZERO here then you will get a division by zero error and the whole task will die. Please make sure it's not zero in this case and if it is skip accordingly perhaps with a log error.